kvirt/k3s/bootstrap.sh

```diff
@@ -1,23 +1,22 @@
 {% set extra_args = [] %}
-{% for component in disabled_components %}
-{% do extra_args.append("--disable " + component) %}
+{% for component in k3s_extra_args_all %}
+{% do extra_args.append(component) %}
+{% endfor %}
+{% for component in k3s_extra_args_masters %}
+{% do extra_args.append(component) %}
 {% endfor %}
 apt-get -y install curl
 {% if masters > 1 %}
-curl -sfL https://get.k3s.io | {{ "INSTALL_K3S_EXEC='--flannel-backend=none'" if sdn != "flannel" else '' }} INSTALL_K3S_CHANNEL={{ install_k3s_channel }} INSTALL_K3S_VERSION={{ install_k3s_version if install_k3s_version != "latest" else '' }} K3S_TOKEN={{ token }} sh -s - server --cluster-init {{ extra_args|join(" ") }}
+curl -sfL https://get.k3s.io | INSTALL_K3S_CHANNEL={{ install_k3s_channel }} INSTALL_K3S_VERSION={{ install_k3s_version if install_k3s_version != "latest" else '' }} K3S_TOKEN={{ token }} sh -s - server --cluster-init {{ extra_args|join(" ") }}
 export IP={{ api_ip }}
 {% else %}
-curl -sfL https://get.k3s.io | {{ "INSTALL_K3S_EXEC='--flannel-backend=none'" if sdn != "flannel" else '' }} INSTALL_K3S_CHANNEL={{ install_k3s_channel }} INSTALL_K3S_VERSION={{ install_k3s_version if install_k3s_version != "latest" else '' }} K3S_TOKEN={{ token }} sh -s - server {{ extra_args|join(" ") }}
+curl -sfL https://get.k3s.io | INSTALL_K3S_CHANNEL={{ install_k3s_channel }} INSTALL_K3S_VERSION={{ install_k3s_version if install_k3s_version != "latest" else '' }} K3S_TOKEN={{ token }} sh -s - server {{ extra_args|join(" ") }}
 export IP=$(hostname -I | cut -f1 -d" ")
 {% endif %}
+apt-get -y remove curl
 export K3S_TOKEN=$(cat /var/lib/rancher/k3s/server/node-token)
 sed "s/127.0.0.1/$IP/" /etc/rancher/k3s/k3s.yaml > /root/kubeconfig
 if [ -d /root/manifests ] ; then
  mkdir -p /var/lib/rancher/k3s/server
  mv /root/manifests /var/lib/rancher/k3s/server
 fi
-{% if sdn == 'cilium' %}
```

Comment:
alternatively, we could fix whatever is missing in cilium side instead of removing it. 
But if you think it doesnt make sense to keep it as is right now, let's instead open a PR to study its removal

Reply:
Yeah agree to the part of your comment that points to the fact that this change shouldn't really be a part of this PR.

In regards to Cilium. It was very much my wish to look into using Cilium instead of flannel. For other users ... with so many choices for CNI's. I think it makes more sense to say:

1. you can leave the CNI as is - meaning what K3s comes with per default
2. or you can uninstall flannel with the K3s install and conf. args and then install whatever CNI you want

At the time, where I wished to get Cilium, I was still trying to grasp how much KCLI should automate for end-users. I guess that is still very much a subject up for debate.